### ISSUE TITLE
user documentation?

### ISSUE THREAD

**Issue by Phillip-M-Feldman** [mrJean1/PyGeodesy#16]
Is there any user documentation for this package?  (I have been unable to find any).

**Comment by mrJean1** [comment]
Other than [this Python](https://mrjean1.github.io/PyGeodesy/) and the [original JavaScript](http://www.movable-type.co.uk/scripts/geodesy/docs/) documentation, this [Geohash](http://www.movable-type.co.uk/scripts/geohash.html), this [LatLon](http://www.movable-type.co.uk/scripts/latlong.html), this [OSGrid](http://www.movable-type.co.uk/scripts/latlong-os-gridref.html), this [UTM-MGRS](http://www.movable-type.co.uk/scripts/latlong-utm-mgrs.html), this [Vector](http://www.movable-type.co.uk/scripts/latlong-vectors.html) and this [Vincenty](http://www.movable-type.co.uk/scripts/latlong-vincenty.html) page and perhaps the [tests](https://github.com/mrJean1/PyGeodesy/tree/master/test), there is not user documentation.

**Comment by Phillip-M-Feldman** [comment]
I withdraw my statement!  There is indeed quite a bit of documentation.
Thanks!

Is it possible to make the "View Docs" button on the
https://readthedocs.org/projects/pygeodesy/ page link to these resources?

Phillip M. Feldman

**Comment by mrJean1** [comment]
Thank you.  Importing and building PyGeodesy from GitHub does not seem to work at Read the Docs.  It is unclear what’s missing or wrong.

**Comment by Phillip-M-Feldman** [comment]
Also, when I go to http://pygeodesy.readthedocs.io/, I get the message
"This page does not yet exist".

**Comment by Phillip-M-Feldman** [comment]
Two more questions: I'm looking at documentation for `Cartesian.toLatLon`.
Given that this method converts an x/y/z point to geodetic, how can LatLon
be an input parameter?  Also, what are the units of x, y, and z?  Meters?
toLatLon(self, datum=Datum(name='WGS84', ellipsoid=Ellipsoids.WGS84,
transform=Tran..., LatLon=<class 'pygeodesy.ellipsoidalNvector.LatLon'>)

Convert this (geocentric) cartesian (x/y/z) point to an (ellipsoidal)
geodetic point on the specified datum.
Parameters:

   - *datum* - Optional datum to use (Datum
   <https://mrjean1.github.io/PyGeodesy/docs/pygeodesy.datum.Datum-class.html>
   ).
   - *LatLon* - Optional LatLon class for the point (LatLon
   <https://mrjean1.github.io/PyGeodesy/docs/pygeodesy.ellipsoidalNvector.LatLon-class.html>
   ).


On Mon, Jun 4, 2018 at 2:11 PM, Phillip Feldman <phillip.m.feldman@gmail.com
> wrote:

> Also, when I go to http://pygeodesy.readthedocs.io/, I get the message
> "This page does not yet exist".
>

**Comment by mrJean1** [comment]
(a) The LatLon argument is a class, not an instance (hence the Capilatized/CamelCase name)!  The default value for that keyword argument is LatLon=ellipsoidalNvector.LatLon, a PyGeodesy class.  So, if you do not specify anything, you’ll get an instance of that LatLon class as the return value.

Assume, you have your own, special subclass of ellipsoidalNvector.LatLon, say

class NeLL(ellipsoidalNvector.LatLon):
    pass

And if you need/want an instance of that NeLL class returned by Cartesian.toLatLon, you use

ll = Cartesian.toLaLon(..., LatLon=NeLL)


(b) The units of x, y and z are the same as the units of major and minor axes of the Datum, usually meter for all predefined Datums in PyGeodesy.  But if you prefer different units, define a private MyDatum class and use that, everywhere as datum=MyDatum.

 
Hope this makes sense?


/Jean

**Comment by mrJean1** [comment]
Yep, there was and there’s still nothing there (other than blank pages).

Tried again later today, but I haven’t figured out why ReadTheDocs has trouble importing and building PyGeodesy (and other projects).

All prequisites are present at GitHub, AFAICT and the RTD builds work, but the PDF pages are blank at RTD.

/Jean
